src/entities/FormattingNode/index.spec.ts

```diff
@@ -0,0 +1,281 @@
+
+      node.insertText(newText, index + offset);
+
+      expect(anotherChildMock.insertText).toBeCalledWith(newText, index);
+    });
+
+    it('should append text to the last child by default', () => {
+      node.insertText(newText);
+
+      expect(anotherChildMock.insertText).toBeCalledWith(newText, anotherChildMock.length);
+    });
+  });
+
+  describe('.removeText()', () => {
+    const start = 3;
+    const end = 5;
+
+    it('should remove text from appropriate child', () => {
+      node.removeText(start, end);
+
+      expect(childMock.removeText).toBeCalledWith(start, end);
+    });
+
+    it('should adjust indices by child offset', () => {
+      const offset = childMock.length;
+
+      node.removeText(offset + start, offset + end);
+
+      expect(anotherChildMock.removeText).toBeCalledWith(start, end);
+    });
+
+    it('should call removeText for each affected child', () => {
+      const offset = childMock.length;
+
+      node.removeText(start, offset + end);
+
+      expect(childMock.removeText).toBeCalledWith(start, offset);
+      expect(anotherChildMock.removeText).toBeCalledWith(0, end);
+    });
+
+    it('should remove all text by default', () => {
+      node.removeText();
+
+      expect(childMock.removeText).toBeCalledWith(0, childMock.length);
+      expect(anotherChildMock.removeText).toBeCalledWith(0, anotherChildMock.length);
+    });
+
+    it('should call remove() if length is 0', () => {
+      const removeSpy = jest.spyOn(node, 'remove');
+      const lengthSpy = jest.spyOn(node, 'length', 'get').mockImplementation(() => 0);
+
+      node.removeText();
+
+      expect(removeSpy).toBeCalled();
+
+      removeSpy.mockRestore();
+      lengthSpy.mockRestore();
+    });
+  });
+
+  describe('.getText()', () => {
+    const start = 3;
+    const end = 5;
+
+    it('should return text from the relevant child', () => {
+      node.getText(start, end);
+
+      expect(childMock.getText).toBeCalledWith(start, end);
+    });
+
+    it('should adjust index by child offset', () => {
+      const offset = childMock.length;
+
+      node.getText(offset + start, offset + end);
+
+      expect(anotherChildMock.getText).toBeCalledWith(start, end);
+    });
+
+    it('should return text from all relevant children', () => {
+      const offset = childMock.length;
+
+      node.getText(start, offset + end);
+
+      expect(childMock.getText).toBeCalledWith(start, offset);
+      expect(anotherChildMock.getText).toBeCalledWith(0, end);
+    });
+
+    it('should return all text by default', () => {
+      node.getText();
+
+      expect(childMock.getText).toBeCalledWith(0, childMock.length);
+      expect(anotherChildMock.getText).toBeCalledWith(0, anotherChildMock.length);
+    });
+  });
+
+  describe('.getFragments()', () => {
+    /**
+     * @todo
+     */
+    it.todo('should return fragments for sub-tree');
+
+    it('should return node\'s fragment', () => {
+      const { fragments } = node.serialized;
```

Comment:
looks like this test if for `.serialized`, but not `.getFragments()` since it never called inside the test